### PR TITLE
Hyprlock ++ Hypridle listeners

### PR DESCRIPTION
**Pull request by kRHYME7**
# Pull Request

## Description

Added Hypridle implementations. 

> [!Important]
> ``` [LOG] Using dbus path /org/freedesktop/login1/session/_32 ```
> Seldomly, some apps don't respect and communicate to inhibit idle! This is not Hyprdots fault or Hypridle
> Therefore if this happens click   🕛  to inhibit idle

### TODO
Should this be an optional package? 

# Hyprlock 

Goals 

1.  Reproduce configs with per monitor scaling compatibility or wait for this to be answered https://github.com/hyprwm/hyprlock/issues/261
2.  Widgets should be modular so users can just source or copy it to there preferred config. see ./.config/hyprlock/widgets 
3. Layouts should be a plug in play configuration see ./.config/hyprlock/presets. This way users can share custom layouts.

### CODE REVIEW

**Review by rubiin — APPROVED** [review]

**Comment by rubiin** [comment]
LGTM looks nice

**Review by rubiin — APPROVED** [review]

**Comment by rubiin** [comment]
I think this can be merged as well

**Comment by rubiin** [comment]
@kRHYME7 I think the merge conflicts are blocking the pr from merge

**Comment by kRHYME7** [comment]
Ohhh I totally forgot about this lol!

**Review by rubiin — APPROVED** [review]
LGTM

File: Configs/.config/hypr/hypridle.conf [hunk 1]
```diff
@@ -0,0 +1,39 @@
+$SCREENLOCK = hyprlock # If you prefer to use swaylock: 'swaylock -f'
```
Comment (prasanthrangan) [inline Configs/.config/hypr/hypridle.conf:1]: I assume this has to go along with hyprlock PR?

Comment (kRHYME7) [reply Configs/.config/hypr/hypridle.conf:1]: Yes sure. Users can reference this PR  for now.

**Comment by kRHYME7** [comment]
Will work on  Hyprlock here

**Comment by ChrolloRifat** [comment]
> Will work on Hyprlock here

okay. The current roadblock we are facing with hyplock is the scaling, right?

**Comment by kRHYME7** [comment]
yes because most of the scaling are in font size

**Comment by dieBakterie** [comment]
When did hyprlock+hypridle gets an option im HyDE to for the User to test?

**Comment by kRHYME7** [comment]
You can test it here or contribute to some layouts.

I literally added the goals. Number 1 is not yet accomplished.

**Comment by dieBakterie** [comment]
Seems Like Goal 1 is reached? Now to the Feedback Sourcing, like with the monitor configs doesn't really worked for me(got removed for a reason), also don't know why the `colors.conf` got deleted and how do i exactly use the `hyprlock.dcol` file? Also could you Change your Script that it's not only for Spotify? I've this weird issue that my Screen locks when executing a Script from HyDE that i guess uses the `globalcontrol.sh`, the wallpaper Change and wallbash mode keybind for example locks my screen, not sure it's propably my fault. And i've this annoying issues that my second Monitor doesn't really wakeup when hypridle is triggered, i need to turn on and off again and then i've the workspaces 2 and 3 where two my second and three my First one ist that's left from my second one. Don't know if that's even Hypridle's, Hyprlock's, Hyprland's fault or my config that's "wrong". I also have the Problem with the wallpaper only schowing up on one Monitor and the other one is black, it's just the Background for hyprlock Not showing. Also `path = screenshot` doesn't work for me. I'm on Nvidia and i don't think it's this PR's fault.

**Comment by se7en-x230** [comment]
I have a bit  special setup for my music
basically I use lf as my file browser and added a keybinding to open mpv with

` mpv --shuffle --no-video --pause=no`

for whatever reason the creation of hypr-mpris.inf works but the extraction to the png only randomly.

If I listen on yt it works fine 

Any ideas?

**Comment by kRHYME7** [comment]
@se7en-x230 ` playerctl metadata --format '{{mpris:artUrl}}' `  this should output the mpris art if it existed.

**Comment by se7en-x230** [comment]
> @se7en-x230 `playerctl metadata --format '{{mpris:artUrl}}'` this should output the mpris art if it existed.

It does that part works fine ... 

I hacked this together and up to now that works 

image

I basically check if the folder has a image (which 99% of them have) and copy it to /tmp/

Not pretty but for now it works

**Comment by kRHYME7** [comment]
Uhm, Can you try the latest commits? I think this script is outdated

**Comment by se7en-x230** [comment]
> Uhm, Can you try the latest commits? I think this script is outdated

Yes that's too much as there are too many changes in several dotfiles for this to work. I wait for your finished dotfiles

**Comment by dieBakterie** [comment]
I've problems with this and the newest `hyprlock.sh` script too.

**Comment by kRHYME7** [comment]
> I've problems with this and the newest hyprlock.sh script too.

Some error message or explanation is better

**Comment by dieBakterie** [comment]
What?

**Comment by se7en-x230** [comment]
Also your roman_clock.sh is still called in one or more of the presets

**Comment by kRHYME7** [comment]
> Also your roman_clock.sh is still called in one or more of the presets

ops, ok will remove it

**Comment by abenezerw** [comment]
IMG_1297
I think if you guys don’t define monitor like eDP-2 blur effect will take on all monitors. Also I don’t get the widgets other than these. Pretty sure there’s a date too.

**Comment by abenezerw** [comment]
> IMG_1297 I think if you guys don’t define monitor like eDP-2 blur effect will take on all monitors. Also I don’t get the widgets other than these. Pretty sure there’s a date too.

Oh and the colors of input field work accordingly for all themes

**Comment by kRHYME7** [comment]
Change wallpapers so it will be updated

**Comment by abenezerw** [comment]
> Change wallpapers so it will be updated

Oh, yeah. it works but i only see that wallpaper for a blink of a sec before it goes full black on lock. i also manually put a sww folder with backgorund png to check and it also shows it for a sec before black bg

**Comment by chrollorifat** [comment]
> > Change wallpapers so it will be updated
> 
> Oh, yeah. it works but i only see that wallpaper for a blink of a sec before it goes full black on lock. i also manually put a sww folder with backgorund png to check and it also shows it for a sec before black bg

it's probably because of the screen resolution of the walllpaper.

**Comment by SyedArbaazHussain** [comment]
> > Change wallpapers so it will be updated
> 
> Oh, yeah. it works but i only see that wallpaper for a blink of a sec before it goes full black on lock. i also manually put a sww folder with backgorund png to check and it also shows it for a sec before black bg


i have hyprlock working properly and also wallpaper synced with the desktop wallpaper

using the below in the hyprlock.conf

path = .cache/hyde/wall.set
OR
path = .cache/hyde/wall.blur

can be used to resolve the background wallpaper issue
the file is symlinked too and gets updated everytime wallpaper changes

**Comment by dieBakterie** [comment]
Yeah funny that this doesn't work with nvidia GPUs both the normal package and the -git version. It's the reason why i've to do it this annoying way.

**Comment by dieBakterie** [comment]
Could we name it hypr.dcol 'cause i need/use it for plugins etc too. The colors in colors.conf arent really colors but the borders and their _colors_ etc. It's more of an hyprecosytem color file than an pure hyprlock one

**Comment by kRHYME7** [comment]
That's an attempt for now it should really be targeted to colors.conf. But as you can see it is used for Group colors. It's not a minor change So I'm afraid to change it on this repo. IRRC, I made a PR for this as I really want to expose all hypr colors

**Comment by dieBakterie** [comment]
> That's an attempt for now it should really be targeted to colors.conf. But as you can see it is used for Group colors. It's not a minor change So I'm afraid to change it on this repo. IRRC, I made a PR for this as I really want to expose all hypr colors

Yeah that's understandable, was afraid of pinging tittu so i pinged you instead, sorry for this khing.

**Comment by rubiin** [comment]
would love to see this, especially the hyridle part

**Comment by rubiin** [comment]
waiting for this . Thats why havent did on my own :D

**Pull request closed without merging**